### PR TITLE
DOP-4614: Skunk - upload lighthouse reports to Atlas

### PR DESCRIPTION
**Pull request by mmeigs**
### Stories/Links:

DOP-4614
Skunkworks project!!

### Links:

[Action runs](https://github.com/mongodb/snooty/actions/runs/9485595091)

### Notes:

This adds the new custom action to Snooty to run lighthouse scores 3 times, average important scores, and upload to Atlas.

This action is run on every merge (push) to main and uploaded to the collection lighthouse.main_reports.

It is also run on every opening and commit of a PR and uploaded to the collection lighthouse.pr_reports.

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

**Review by mayaraman19 — COMMENTED** [review]
this is cool! i have a couple questions/comments:

- for my own knowledge, what is the difference between/uses for the `jsonRuns` and `htmlRuns` objects in the documents in Atlas?
- for `commitMessage` and `commitTimestamp` in Atlas, I only see a bunch corresponding to the "new summary properties" commit, is this expected?
- also, it is really cool/convenient to be able to access this data in Atlas, but as we talked about briefly in slack it does take a long time for the data to show up. wondering if the `htmlRuns`/`jsonRuns` objects should be moved to AWS or something, or maybe it's fine? just wanted to note it down here and perhaps the team can discuss.

**Comment by mmeigs** [comment]
> this is cool! i have a couple questions/comments:
> 
> * for my own knowledge, what is the difference between/uses for the `jsonRuns` and `htmlRuns` objects in the documents in Atlas?
> * for `commitMessage` and `commitTimestamp` in Atlas, I only see a bunch corresponding to the "new summary properties" commit, is this expected?
> * also, it is really cool/convenient to be able to access this data in Atlas, but as we talked about briefly in slack it does take a long time for the data to show up. wondering if the `htmlRuns`/`jsonRuns` objects should be moved to AWS or something, or maybe it's fine? just wanted to note it down here and perhaps the team can discuss.

Great questions! 
1. Each Lighthouse run gives us a JSON object with all the data gleaned from the run, but it also gives us an HTML string that shows a pretty markup of the scores of the run. For now, I thought it would be great to have access to both so that if our displays (using the JSON data) are not showing you the metrics you need, you can click through to the full HTML report.
2. As far as quite a few having the same commitMessage, that is correct. Each commit in a Snooty PR (or on main) will create 12 documents. Lighthouse will be run on 2 separate projects (cloud-docs and server docs), and each project runs Lighthouse on 3 urls, and each url will get two Lighthouse runs (once for mobile and once for desktop). A lot of numbers but basically, it creates 12 documents (2 * 3 * 2) that all have the same commitMessage.
3. Discussed offline, we will upload to Atlas for now. But we can change this in the future!! (also on the table, we could not upload the HTML runs if we decide they are unnecessary)

**Review by mmeigs — COMMENTED** [review]
@seungpark @anabellabuckvar @mayaraman19 
This is finally ready to be re-reviewed!! Everything works! Hallelu. 

Feel free to check Atlas. There are two full runs (commits) in there. And feel free to reach out with any questions!

**Review by mayaraman19 — APPROVED** [review]
beautiful! although i think this should only be merged after 1 more approval. your commit messages read like poetry 😽

File: .github/workflows/lighthouse.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+on:
+  push:
+    branches:
+      - "main"
+  pull_request:
+    types: ["opened", "synchronize"]
```
Comment (seungpark) [inline .github/workflows/lighthouse.yml:6]: do we want to store to DB on each PR? given the size, how is on each commit to main

File: .github/workflows/lighthouse.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          GATSBY_PARSER_USER: docsworker-xlarge
+          GATSBY_SITE: ${{ matrix.project }}
+          GATSBY_PARSER_BRANCH: main
+        run: |
+          npm ci --legacy-peer-deps
+          npm run build
+      - name: Run Lighthouse
+        env: 
+          GATSBY_SITE: ${{ matrix.project }}
+          LIGHTHOUSE_SERVER_URL: ${{ secrets.LIGHTHOUSE_SERVER_URL }}
```
Comment (seungpark) [inline .github/workflows/lighthouse.yml:39]: minor. do we need these LIGHTHOUSE_ constants anymore?

**Review by seungpark — APPROVED** [review]
minor comments below but looks good! thank you @mmeigs for doing this!

File: .github/workflows/lighthouse.yml (hunk 2 above)
Comment (mmeigs) [reply .github/workflows/lighthouse.yml:39]: ^^ great callout. We do not!

File: .github/workflows/lighthouse.yml (hunk 1 above)
Comment (mmeigs) [reply .github/workflows/lighthouse.yml:6]: Yes? I think this is the whole reason. So, that we can know when developing (before merge) how we are affecting scores.
Also, I have a TTL index on the collection `pr_reports`.

Comment (mmeigs) [reply .github/workflows/lighthouse.yml:6]: I think a possible future ticket could even be a deletion of all the PR's lighthouse builds when the PR is closed!

**Pull request merged**